lib/tokenizers/decode_stream.ex

```diff
@@ -0,0 +1,56 @@
+defmodule Tokenizers.DecodeStream do
+  @moduledoc """
+  Implements streaming decoding functionality for tokenizers.
+  """
+
+  @enforce_keys [:resource]
+  defstruct [:resource]
+
+  @type t :: %__MODULE__{
+          resource: reference()
+        }
+
+  @doc """
+  Creates a new decode stream.
+
+  The `skip_special_tokens` option determines whether special tokens should be skipped during decoding.
+  By default, it is set to `false`.
+  """
+  @spec new(boolean()) :: t()
+  def new(skip_special_tokens \\ false) do
+    Tokenizers.Native.decoder_stream_new(skip_special_tokens)
+  end
```

Comment:
Let's make the argument keyword options, so it is easy to expand if new options are added in the future.

```suggestion
  ## Options

    * `:skip_special_tokens` - determines whether special tokens should be
      skipped during decoding. By default, it is set to `false`.

  """
  @spec new(keyword()) :: t()
  def new(opts \\ []) when is_list(opts) do
    opts = Keyword.validate!(skip_special_tokens: false)
    Tokenizers.Native.decoder_stream_new(opts[:skip_special_tokens])
  end
```